src/engine/PhotoStudio.cs

```diff
@@ -451,4 +464,12 @@ private Node3D CreateNewRoot(IWorldSimulation worldSimulation)
 
         return node;
     }
+
+    private class TaskComparer : IComparer<(int, int)>
+    {
+        public int Compare((int, int) x, (int, int) y)
+        {
+            return x.CompareTo(y);
```

Comment:
I tried to find the memory allocations using benchmark dot net, but I didn't see anything. Though that isn't conclusive as I was unable to get it to detect memory usage when I intentionally added stuff.

For maximum clarity in this critical system, I'd prefer the following a bit more verbose approach here:

```suggestion
            if (x.Item1 < y.Item1)
                return -1;

            if (x.Item1 > y.Item1)
                return 1;

            return x.Item2.CompareTo(y.Item2);
```